ted2zim/scraper.py

```diff
@@ -1,129 +1,145 @@
-            sys.exit("zimwriterfs is not available, please install it.")
 
-    def extract_page_number(self):
-        """
-        Extract the number of video pages by looking at the
-        pagination div at the bottom. Select all <a>-tags in it and
-        return the last element in the list. That's our total count
-        """
-        self.soup = BeautifulSoup(utils.download_from_site(self.BASE_URL).text)
-        pages = self.soup.select("div.pagination a.pagination__item")[-1]
-        return int(pages.text)
+        self.debug = debug
+
+    @property
+    def root_dir(self):
+        return ROOT_DIR
+
+    @property
+    def templates_dir(self):
+        return self.root_dir.joinpath("templates")
+
+    @property
+    def build_dir(self):
+        return self.output_dir.joinpath("build")
+
+    @property
+    def ted_videos_json(self):
+        return self.output_dir.joinpath("ted_videos.json")
+
+    @property
+    def ted_categories_json(self):
+        return self.output_dir.joinpath("ted_categories.json")
 
     def extract_all_video_links(self):
-        """
-        This method will build the specifiv video site by appending
-        the page number to the 'page' parameter to the url.
-        We will iterate through every page and extract every
-        video link. The video link is extracted in `extract_videos()`.
-        """
-        for page in range(1, self.extract_page_number()):
-            url = utils.build_video_page(page)
-            html = utils.download_from_site(url).text
-            self.soup = BeautifulSoup(html)
-            self.extract_videos()
-            print("Finished scraping page " + str(page))
-
-    def extract_videos(self):
-        """
-        All videos are embedded in a <div> with the class name 'row'.
-        We are searching for the div inside this div, that has an <a>-tag
-        with the class name 'media__image', because this is the relative
-        link to the representative TED talk. We have to turn this relative
-        link to an absolute link. This is done through the `utils` class.
-        """
-        print(
-            "Video found : " + str(len(self.soup.select("div.row div.media__image a")))
-        )  # DEBUG
-        for video in self.soup.select("div.row div.media__image a"):
-            url = utils.create_absolute_link(self.BASE_URL, video["href"])
+
+        # extracts all video links for different categories
+        # it iterates over the categories and then over pages to get required number of video links
+        for category in self.categories:
+            logger.debug(f"Fetching video links for category: {category}")
+            category_url = f"{self.BASE_URL}?topics%5B%5D={category}"
+            self.soup = BeautifulSoup(
+                download_from_site(category_url).text, features="html.parser"
+            )
+            video_allowance = None
+            page = 1
+            tot_videos_scraped = 0
+            if self.max_videos == "max":
```

Comment:
this value should be for end-users only. If that's what the user wants, then set this value earlier.

Reply:
Its simply `video_allowance = self.max_videos_per_topic` now.